### DISCUSSION TITLE
Drag and drop support

### DISCUSSION THREAD

**Discussion by poisa** [NativePHP/laravel#279] [Ideas]
It would be great to be able to drag files into the app and drag files out of the app.

**Comment by thomas-alrek** [comment]
+1

**Comment by koossaayy** [comment]
I believe you can drag into the app, like any other drag. But the drag out of the app, I don't think that's possible for the moment. And correct me if I am wrong 😅

**Comment by simonhamp** [comment]
Drag-and-drop _in_ to your app is supported inherently by virtue of using a web view... how you handle that is the same way you'd handle it in a traditional web app, with some front-end Javascript code.

Drag-and-drop _out_ of your app requires a little bit of set up and coordination between Electron and the front-end of your application. [This will need implementing](https://www.electronjs.org/docs/latest/tutorial/native-file-drag-drop) in the [`nativephp/electron-plugin`](https://github.com/NativePHP/electron-plugin/).

**Reply by poisa** [reply]
> Drag-and-drop in to your app is supported inherently by virtue of using a web view... how you handle that is the same way you'd handle it in a traditional web app, with some front-end Javascript code.

I'm not sure how to "connect" the dropped in files with the PHP layer to be honest, but for my use case, this is all I need. I can certainly accept drag into the app for my inputs and then create files without the need to drag out. I'll try to figure this out and if all fails I'll ask the GPT gods for help.

Thanks for making this all open source, by the way!
